crawler4j-core/src/main/java/edu/uci/ics/crawler4j/crawler/dynamic/DynamicCrawlerHelper.java

```diff
@@ -0,0 +1,67 @@
+package edu.uci.ics.crawler4j.crawler.dynamic;
+
+import crawlercommons.filters.basic.BasicURLNormalizer;
+import edu.uci.ics.crawler4j.crawler.CrawlController;
+import edu.uci.ics.crawler4j.crawler.DynamicCrawlConfig;
+import edu.uci.ics.crawler4j.crawler.Page;
+import edu.uci.ics.crawler4j.fetcher.PageFetcher;
+import edu.uci.ics.crawler4j.frontier.FrontierConfiguration;
+import edu.uci.ics.crawler4j.parser.dynamic.DynamicParser;
+import edu.uci.ics.crawler4j.robotstxt.RobotstxtConfig;
+import edu.uci.ics.crawler4j.robotstxt.RobotstxtServer;
+import edu.uci.ics.crawler4j.url.TLDList;
+import org.openqa.selenium.By;
+
+import java.util.function.BiFunction;
+
+/**
+ * Helper class that provides factory methods for a {@link CrawlController} set up for headless browser crawling.
+ */
+public class DynamicCrawlerHelper {
```

Comment:
What do you think about introducing a new module `crawler4j-extensions` which contains `crawler4j-extensions-dynamic` (not sure about the names) to encapsulate everything related to dynamic scraping? That would make it possible to be lightweight or getting the full dynamic crawling capacity? wdyt?